### ISSUE TITLE
Consider using pako as a zlib.js implementation

### ISSUE THREAD

**Issue by baldurmen** [glowing-bear/glowing-bear#1082]
Hi!

While working on the Debian package, I had a look at how the compression is done. At the moment, a minimized file (that is not properly copyrighted btw :P) in `3rdparty/inflate.min.js` is loaded.

[It seems](https://github.com/glowing-bear/glowing-bear/issues/88) this file comes from [the zlib.js project](https://github.com/imaya/zlib.js).

It also seems nowadays people are using [pako](https://github.com/nodeca/pako) for an implementation of zlib in JS, as apparently it's faster.

pako happens to be packaged in Debian and I'm likely to try to patch `js/weechat.js` to use that instead. I'll try to send a patch when I have something working.

**Comment by lorenzhs** [comment]
Yeah it's from zlib.js. The file starts with `/** @license zlib.js 2012 - imaya [ https://github.com/imaya/zlib.js ] The MIT License */`, which is I think enough to comply with the license.

Speed is rather irrelevant for us, there's not a lot of data going over the pipe. But pako's minified file is more than 3x the size (22 vs 7kb). I'm not convinced that there's a good argument to switch tbqh.

**Comment by AStove** [comment]
The browser should do this. See [my request](https://github.com/weechat/weechat/issues/1549) to have weechat changed so it supports this websocket extension that takes care of compression on the browser level.

**Comment by baldurmen** [comment]
Well, for now [I've patched it in Debian](https://sources.debian.org/src/glowing-bear/0.9.0+ds-1/debian/patches/0001_Use_Packaged_JS_Dependencies.patch/#L66) and it seems to work pretty well :) (oh and glowing-bear should make it in the next Debian stable release \0/)

> Would a debian package still use a browser? If not, this might be off topic.

It would yes!
